src/components/CodeManagement/ManageRequestsTab.jsx

```diff
@@ -44,17 +73,57 @@ const ManageRequestsTab = ({ enterpriseId }) => {
           pageSize: PAGE_SIZE,
           pageIndex: 0,
         }}
+        disableApproveButton={!hasAvailableCodes}
       />
+      {selectedRequest && (
+        <>
+          {isApproveModalOpen && (
+            <ApproveCouponCodeRequestModal
+              isOpen
+              couponCodeRequest={selectedRequest}
+              onSuccess={() => {
+                updateRequestStatus({ request: selectedRequest, newStatus: SUBSIDY_REQUEST_STATUS.PENDING });
+                setIsApproveModalOpen(false);
+              }}
+              onClose={() => setIsApproveModalOpen(false)}
+            />
+          )}
+          {isDenyModalOpen && (
+            <DeclineSubsidyRequestModal
+              isOpen
+              subsidyRequest={selectedRequest}
+              declineRequestFn={EnterpriseAccessApiService.declineCouponCodeRequests}
+              onSuccess={() => {
+                updateRequestStatus({ request: selectedRequest, newStatus: SUBSIDY_REQUEST_STATUS.DECLINED });
```

Comment:
The `onSuccess` and `onClose` callbacks might be worth covering in test(s) to ensure they are doing the correct things related to updating request status, etc. (similar feedback for the subscription requests)? 🤔  Currently, the modals are mocked in the tests for these components, but perhaps we could make the mocked components call `onSuccess` / `onClose` to get coverage on these callbacks?